### DISCUSSION TITLE
Feature request: Flatpak support

### DISCUSSION THREAD

**Discussion by Pshemas** [mu-editor/mu#1584] [Ideas & New Features]
I'd love to see Flatpak support for Mu . It would make installing on Linux, no matter the distro, much easier and less messy , especially in educational environments (which is what I'm interested in :) ).
Alternatively appimage would be cool, but in education managing flatpaks is way more straightforward.

**Comment by ntoll** [comment]
cc/ @tmontes

**Comment by jwillikers** [comment]
It would be much easier for me to use Mu on Fedora Silverblue if it was packaged as a Flatpak. I'd love to see this!

**Comment by carlosperate** [comment]
> Alternatively appimage would be cool, but in education managing flatpaks is way more straightforward.

It'd be really interesting to understand why flatpaks are more straightforward to manage in education? As a network administrator there is an easy way to centrally push an install of a flatpak in all systems?

I personally favour the simplicity of appimage, simply download a file and execute it. Which is less steps than having to make sure users have flatpak to begin with. Is there any significant advantage in education for flatpak?

**Reply by jwillikers** [reply]
> > Alternatively appimage would be cool, but in education managing flatpaks is way more straightforward.
> 
> It'd be really interesting to understand why flatpaks are more straightforward to manage in education? As a network administrator there is an easy way to centrally push an install of a flatpak in all systems?
> 

An educational institution can host their own Flatpak repository, allowing them to control default permission and which applications are available. Since auto-updates are built-in to Flatpak, they will be automatically updated from said repository when updates are made available.

You can probably do the same with AppImages, technically, but most everything is already baked in to how Flatpak works. The AppImage ecosystem is definitely improving. I see more AppImage app stores popping up and work on sandboxing. The main thing is you have to grab all the various projects for AppImage to get similar functionality to Flatpak. One annoyance I face with AppImages is desktop integration, where I have to install another tool to easily integrate them, i.e. [AppImageLauncher](https://github.com/TheAssassin/AppImageLauncher).

> I personally favour the simplicity of appimage, simply download a file and execute it. Which is less steps than having to make sure users have flatpak to begin with. Is there any significant advantage in education for flatpak?

The simplicity of AppImage is a bit of a double-edged sword there. It is simpler in that way when the user doesn't have Flatpak installed, but it doesn't ensure the AppImage came from a trusted source.

I'll add that discoverability of AppImages isn't quite as competitive as Flatpak, or snaps for that matter, as a lot of users discover software through graphical app stores or software centers. Most major distributions ship with software centers that support system packages, and additionally Flatpaks and/or snaps.

**Comment by tmontes** [comment]
All,

Without any disregard for the requested FlatPak or any of the notes in the discussion, I'm sharing a first attempt at AppImage Linux packaging for the current release of Mu: please grab it from [the bottom of this page](https://github.com/mu-editor/mu/actions/runs/2024827478).

Just tested it on Debian 10:
* Download.
* Make executable.
* Run it!

Let me know how it goes, please! :)
